frontend/packages/dev-console/src/actions/providers.ts

```diff
@@ -116,124 +123,151 @@ export const useTopologyGraphActionProvider: TopologyActionProvider = ({
   const isSampleTypeEnabled = isCatalogTypeEnabled(SAMPLE_CATALOG_TYPE_ID);
 
   return React.useMemo(() => {
+    if (!isGraph(element)) {
+      return [[], true, undefined];
+    }
+
     const sourceObj = connectorSource?.getData()?.resource;
     const sourceReference = sourceObj
       ? `${referenceFor(sourceObj)}/${sourceObj?.metadata?.name}`
       : undefined;
 
-    const actionsWithSourceRef: Action[] = [];
-    actionsWithSourceRef.push(
-      AddActions.FromGit(namespace, undefined, sourceReference, '', !isImportResourceAccess),
-    );
-    actionsWithSourceRef.push(
-      AddActions.ContainerImage(
-        namespace,
-        undefined,
-        sourceReference,
-        '',
-        !isCatalogImageResourceAccess,
-      ),
-    );
-    if (isOperatorBackedServiceEnabled) {
-      actionsWithSourceRef.push(
-        AddActions.OperatorBacked(
-          namespace,
-          undefined,
-          sourceReference,
-          '',
-          null,
-          serviceBindingAccess,
-        ),
-      );
-    }
-    actionsWithSourceRef.push(
-      AddActions.UploadJarFile(
-        namespace,
-        undefined,
-        sourceReference,
-        '',
-        !isCatalogImageResourceAccess,
-      ),
-    );
+    const actions: Action[] = [];
 
-    const actionsWithoutSourceRef: Action[] = [];
-    if (isSampleTypeEnabled) {
-      actionsWithoutSourceRef.push(
-        AddActions.Samples(
-          namespace,
-          undefined,
-          undefined,
-          ADD_TO_PROJECT,
-          !isImportResourceAccess,
-        ),
-      );
-    }
-    actionsWithoutSourceRef.push(
-      AddActions.FromGit(namespace, undefined, undefined, ADD_TO_PROJECT, !isImportResourceAccess),
-    );
-    actionsWithoutSourceRef.push(
-      AddActions.ContainerImage(
-        namespace,
-        undefined,
-        undefined,
-        ADD_TO_PROJECT,
-        !isCatalogImageResourceAccess,
-      ),
-    );
-    if (isDevCatalogEnabled) {
-      actionsWithoutSourceRef.push(
-        AddActions.DevCatalog(namespace, undefined, undefined, ADD_TO_PROJECT, undefined),
-      );
-      actionsWithoutSourceRef.push(
-        AddActions.DatabaseCatalog(namespace, undefined, undefined, ADD_TO_PROJECT, undefined),
-      );
+    if (sourceReference) {
+      if (openshiftFlag) {
+        actions.push(
```

Comment:
Nit: We can pass multiple elements to [Array.push](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) at once.

Reply:
Good point, improved it (3 times), first force-push contains just this change, second force-push is a rebase